aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1348 @@
+    use aptos_std::table::{Self, Table};
+    use std::bcs::to_bytes;
+    use std::error;
+    use std::hash::sha3_256;
+    use std::option::{Self, Option};
+    use std::signer::address_of;
+    use std::string::String;
+    use std::vector;
+
+    /// The salt used to create a resource account during multisig account creation.
+    /// This is used to avoid conflicts with other modules that also create resource accounts with the same owner
+    /// account.
+    const DOMAIN_SEPARATOR: vector<u8> = b"aptos_framework::multisig_account";
+
+    // Any error codes > 2000 can be thrown as part of transaction prologue.
+    /// Owner list cannot contain the same address more than once.
+    const EDUPLICATE_OWNER: u64 = 1;
+    /// Specified account is not a multisig account.
+    const EACCOUNT_NOT_MULTISIG: u64 = 2002;
+    /// Account executing this operation is not an owner of the multisig account.
+    const ENOT_OWNER: u64 = 2003;
+    /// Transaction payload cannot be empty.
+    const EPAYLOAD_CANNOT_BE_EMPTY: u64 = 4;
+    /// Multisig account must have at least one owner.
+    const ENOT_ENOUGH_OWNERS: u64 = 5;
+    /// Transaction with specified id cannot be found.
+    const ETRANSACTION_NOT_FOUND: u64 = 2006;
+    /// Cannot execute the specified transaction simply via transaction_id as the full payload is not stored on chain.
+    const EPAYLOAD_NOT_STORED: u64 = 7;
+    /// Provided target function does not match the hash stored in the on-chain transaction.
+    const EPAYLOAD_DOES_NOT_MATCH_HASH: u64 = 2008;
+    /// Transaction has not received enough approvals to be executed.
+    const ENOT_ENOUGH_APPROVALS: u64 = 2009;
+    /// Transaction has not received enough rejections to be removed.
+    const ENOT_ENOUGH_REJECTIONS: u64 = 10;
+    /// Number of signatures required must be more than zero and at most the total number of owners.
+    const EINVALID_SIGNATURES_REQUIRED: u64 = 11;
+    /// Payload hash must be exactly 32 bytes (sha3-256).
+    const EINVALID_PAYLOAD_HASH: u64 = 12;
+    /// The multisig account itself cannot be an owner.
+    const EOWNER_CANNOT_BE_MULTISIG_ACCOUNT_ITSELF: u64 = 13;
+    /// Owner has already approved this transaction before.
+    const TRANSACTION_HAS_ALREADY_BEEN_APPROVED: u64 = 14;
+    /// Owner has already rejected this transaction before.
+    const TRANSACTION_HAS_ALREADY_BEEN_REJECTED: u64 = 15;
+
+    /// Indicates the transaction is still pending approvals or rejections.
+    const TRANSACTION_STATUS_PENDING: u64 = 1;
+    /// Indicates the transaction has been executed successfully.
+    const TRANSACTION_STATUS_SUCCEEDED: u64 = 2;
+    /// Indicates the transaction execution has failed. The transaction would still be considered resolved and cannot
+    /// be re-executed.
+    const TRANSACTION_STATUS_FAILED: u64 = 3;
+    /// Transaction has received enough rejections from owners and been removed/skipped.
+    const TRANSACTION_STATUS_REMOVED: u64 = 4;
```

Comment:
we should reclaim the storage when remove it.

Reply:
It's tricky because we want to keep the error when execution of the target transaction fails (it's not stored anywhere in storage because the overall transaction still succeeds).

Reply:
Any suggestion for better alternatives with your expert understanding of storage?

Reply:
Updated to not store transactions after execution.